modules/opensearch/opensearch.module

```diff
@@ -31,6 +31,190 @@ function opensearch_ding_provider() {
   );
 }
 
+/**
+ * Implements hook_theme().
+ */
+function opensearch_theme() {
+  return array(
+    'opensearch_ranking_field' => array(
+      'arguments' => array('element' => NULL),
+    ),
+  );
+}
+
+/**
+ * Implements hook_element_info().
+ *
+ * Define a new form element named opensearch_ranking_field.
+ *
+ * It calls the theme function theme_opensearch_ranking_field, if not defined
+ * no elements will be displayed.
+ */
+function opensearch_element_info() {
+  return array(
+    'opensearch_boost_field' => array(
+      '#input' => TRUE,
+      '#process' => array('opensearch_boost_field_element_process'),
+    ),
+    'opensearch_ranking_field' => array(
+      '#input' => TRUE,
+      '#process' => array('opensearch_ranking_field_element_process'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_permission().
+ */
+function opensearch_permission() {
+  return array(
+    'administer opensearch settings' => array(
+      'title' => t('Administer Opensearch settings'),
+    ),
+  );
+}
+
+/**
+ * Implements hook_menu().
+ */
+function opensearch_menu() {
+  $items['admin/config/opensearch'] = array(
```

Comment:
I am wondering where is the right place to put this.

- Should it be under `admin/config/ting/opensearch` to group it with related elements?
- How does it relate to the provider settings form `opensearch_admin_settings`. That form is currently being shown under the provider settings. One approach could be to also use local tasks under `admin/config/ting/opensearch` and then use that as the default task.
- This would also provide a better home for the autocomplete settings in #82